scripts/exgfs_chem_init_aerosol.py

```diff
@@ -84,17 +87,55 @@ def get_env_var(varname: str, fail_on_missing: bool = True) -> str:
 def get_atm_files(path: str) -> typing.List[str]:
 	print(f'Checking for atm files in {path}')
 
-	files = list(map(lambda tile: atm_file_pattern.format(tile=tile, path=path), tiles))
-	for file_name in files:
+	file_list = []
+	for file_pattern in atm_file_pattern, atm_ctrl_pattern:
+		files = list(map(lambda tile: file_pattern.format(tile=tile, path=path), tiles))
+		for file_name in files:
+			if(debug):
+				print(f"\tChecking for {file_name}")
+			if(not os.path.isfile(file_name)):
+				print("FATAL: Atmosphere file {file_name} not found".format(file_name=file_name))
+				exit(101)
+			elif(debug):
+				print(f"\t\tFound {file_name}")
+		file_list = file_list + [ files ]
+	return file_list
+
+# Find last cycle with restart files available
+def get_restart_files(time: datetime, incr: int, max_lookback: int, fcst_length: int, rot_dir: str, cdump: str) -> typing.List[str]:
+	print(f"Looking for restart tracer files in {rot_dir}")
+	for lookback in map(lambda i: incr * (i + 1), range(max_lookback)):
+		if(lookback > fcst_length):
+			# Trying to look back farther than the length of a forecast
+			break
+		elif(lookback == fcst_length):
+			# Restart files at the end of the cycle don't have a timestamp
+			timestamp = ""
+		else:
+			timestamp = time.strftime("%Y%m%d.%H0000.")
+
+		last_time = time - timedelta(hours=lookback)
+
 		if(debug):
-			print(f"\tChecking for {file_name}")
-		if(not os.path.isfile(file_name)):
-			print("FATAL: Atmosphere file {file_name} not found".format(file_name=file_name))
-			exit(101)
-		elif(debug):
-			print(f"\t\tFound {file_name}")
-	return files
+			print(f"\tChecking {last_time}")
+		file_list = []
+		file_base = last_time.strftime(restrt_base_pattern.format(**locals()))
+		for file_pattern in tracer_file_pattern, restrt_file_pattern, dycore_file_pattern:
+			files = list(map(lambda tile: file_pattern.format(timestamp=timestamp, file_base=file_base, tile=tile), tiles))
+			if(debug):
+				print(f"\t\tLooking for files {files} in directory {file_base}")
+			found = [file for file in files if os.path.isfile(file)]
+			if(found):
+				file_list = file_list + [ files ]
+			else:
+				print(last_time.strftime("Restart files not found for %Y%m%d_%H"))
+				break
 
+		if(found):
+			return file_list
+		else:
+			print("WARNING: Unable to find restart files, will use zero fields")
+			return None
 
 # Find last cycle with tracer data available via restart files
 def get_tracer_files(time: datetime, incr: int, max_lookback: int, fcst_length: int, rot_dir: str, cdump: str) -> typing.List[str]:
```

Comment:
If get_tracer_files() isn't going to be used anymore, it should be deleted (these changes already break the function, since tracer_base_pattern is no longer defined).

Reply:
Commit 754324d547ad26a9486cb58fe09c882c8138c438 removed `get_tracer_files()`.